src/polyhedralGravity/model/TreeNodeFactory.cpp

```diff
@@ -0,0 +1,14 @@
+#include "TreeNodeFactory.h"
+#include "KDTree.h"
+#include "LeafNode.h"
+#include "SplitNode.h"
+
+namespace polyhedralGravity {
+    std::unique_ptr<TreeNode> TreeNodeFactory::treeNodeFactory(const SplitParam &splitParam) {
+        auto [plane, planeCost, triangleIndexLists] = KDTree::findPlane(splitParam);                                //find optimal plane splitting this node's bounding box
+        if (planeCost > static_cast<double>(splitParam.indexBoundFaces.size()) * KDTree::triangleIntersectionCost) {//if the cost of splitting this node further is greater than just traversing the bound triangles, then don't split and return a LeafNode
```

Comment:
Long `if` statement. Better refactor the multiplication on the right to a `const` variable